src/main/java/com/google/api/generator/gapic/composer/ClassComposer.java

```diff
@@ -16,9 +16,13 @@
 
 import com.google.api.generator.gapic.model.GapicClass;
 import com.google.api.generator.gapic.model.Message;
+import com.google.api.generator.gapic.model.ResourceName;
 import com.google.api.generator.gapic.model.Service;
 import java.util.Map;
 
 public interface ClassComposer {
-  GapicClass generate(Service service, Map<String, Message> messageTypes);
+  // TODO(b/174257081): Clean up the hierarchy to avoid pass another parameter (resourceNames is
+  // only used for composing sample code).
+  GapicClass generate(
+      Service service, Map<String, Message> messageTypes, Map<String, ResourceName> resourceNames);
```

Comment:
As mentioned [here](https://github.com/googleapis/gapic-generator-java/pull/512#discussion_r530645801) and [here](https://github.com/googleapis/gapic-generator-java/pull/512#discussion_r528046098), let's not put `resourceNames` in the interface.

Reply:
Remove the resource name from `ClassComposer` interface.
Move `TODO` under `GapicClass`

Used `TODO(username)` instead.